spec/unit/models/runtime/task_model_spec.rb

```diff
@@ -214,41 +214,26 @@ module VCAP::CloudController
           end
         end
 
-        context 'when the droplet execution metadata specifies a user' do
-          it 'returns the specified user' do
-            expect(task.run_action_user).to eq('some-user')
-          end
-        end
-
-        context 'when the droplet execution metadata DOES NOT specify a user' do
-          let(:droplet_execution_metadata) { '{"entrypoint":["/image-entrypoint.sh"]}' }
-
-          it 'defaults the user to root' do
-            expect(task.run_action_user).to eq('root')
-          end
-        end
-
-        context 'when the droplet execution metadata is an empty string' do
-          let(:droplet_execution_metadata) { '' }
+        context 'when the task DOES NOT have a user specified' do
+          context 'when there is a droplet and it has the docker lifecycle' do
+            before do
+              allow(task.droplet).to(receive(:docker_user).and_return('DropletDockerUser'))
+            end
 
-          it 'defaults the user to root' do
-            expect(task.run_action_user).to eq('root')
+            it 'returns the docker_user from the droplet' do
+              expect(task.run_action_user).to eq('DropletDockerUser')
+            end
           end
         end
 
-        context 'when the droplet execution metadata is nil' do
-          let(:droplet_execution_metadata) { nil }
-
-          it 'defaults the user to root' do
-            expect(task.run_action_user).to eq('root')
+        context 'when there is no droplet for the task' do
+          before do
+            task.droplet.delete
+            task.reload
           end
-        end
-
-        context 'when the droplet execution metadata has invalid json' do
-          let(:droplet_execution_metadata) { '{' }
 
-          it 'defaults the user to root' do
-            expect(task.run_action_user).to eq('root')
+          it 'returns the default "vcap" user' do
+            expect(task.run_action_user).to eq('vcap')
```

Comment:
Why is the default changing here? Is this because of the `allow_docker_root_user` config used in tests? 

I wouldn't expect this to change, since the default should be the existing behavior (`root` is allowed, and `root` is the default user).